### ISSUE TITLE
Confusing error with wrong number of arguments and double splat

### ISSUE THREAD

**Issue by jwoertink** [crystal-lang/crystal#13410]
This isn't a bug, but more of an error message I think could be touched up

```crystal
def headers(**values)
end

headers("Content-Type", "application/json")
```

This code throws the compile error:

```
Showing last frame. Use --error-trace for full trace.

error in line 4
Error: wrong number of arguments for 'headers' (given 2, expected 0)

Overloads are:
 - headers(**values)
 ```

Where is says `(given 2, expected 0)` sort of leads to some confusion here since it's more like "expected 0 or 1".

This code can be fixed by replacing the `, ` with a `: `...

```crystal
def headers(**values)
end

headers("Content-Type": "application/json")
```

❯ crystal -v
Crystal 1.8.1 [a59a3dbd7] (2023-04-20)

LLVM: 15.0.7
Default target: x86_64-unknown-linux-gnu

**Comment by straight-shoota** [comment]
Yeah, this message isn't ideal.

Why "expected 0 or 1" though? The method does not accept any positional arguments, only named ones. But of those you can have as many as you like.

Maybe the error message could say "wrong number of positional arguments" for more clarity?

**Comment by jwoertink** [comment]
Good point. I guess I was just seeing it as a single argument, but now that I think about it, even though it's a NamedTuple, you can't pass in with the curly braces. I think "wrong number of positional arguments" makes it a bit more clear :+1:
